packages/core/src/plugins/resolve.ts

```diff
@@ -36,6 +42,13 @@ function applyExtensions({ chain }: { chain: RspackChain }) {
   ];
 
   chain.resolve.extensions.merge(extensions);
+
+  if (tsconfigPath) {
+    // TypeScript allows importing TS files with `.js` extension
+    chain.resolve.extensionAlias.merge({
+      '.js': ['.ts', '.tsx', '.js'],
```

Comment:
Should we also add `'.jsx': ['.jsx', '.tsx']` as the default `extensionAlias`?

Reply:
Agree, we can add `'.jsx': ['.tsx', '.jsx']` to align with TypeScript. TS resolves `.tsx` first.